### DISCUSSION TITLE
when new Alert Destination email  , Unable to set email content, Only set Subject Template ??????

### DISCUSSION THREAD

**Discussion by ZhangYang19980825** [getredash/redash#7146] [Q&A]
when new Alert Destination email  , Unable to set email content, Only set Subject Template ??????

**Comment by zachliu** [comment]
yes, when setting up a new alert destination, only the subject is configurable
if you need to configure the email body, you need to edit the alert itself

**Comment by ZhangYang19980825** [comment]
but alert  itself,  I didn’t see the location where I can set the content. Could it be this name?
[cid:adc15423-99b6-42d4-8c4e-9d92c728b298]
________________________________
发件人: Zach Liu ***@***.***>
发送时间: 2024年9月9日 23:00
收件人: getredash/redash ***@***.***>
抄送: Yang Zhang @ Marsladder ***@***.***>; Author ***@***.***>
主题: Re: [getredash/redash] when new Alert Destination email , Unable to set email content, Only set Subject Template ?????? (Discussion #7146)


yes, when setting up a new alert destination, only the subject is configurable
if you need to configure the email body, you need to edit the alert itself

―

**Reply by ZhangYang19980825** [reply]
I understand, I am using the v8 version， Should I upgrade the version to v10 ?

**Reply by zachliu** [reply]
ah right, i'm using the latest `24.09.0-dev`

**Reply by ZhangYang19980825** [reply]
Thanks from across the ocean

**Comment by ZhangYang19980825** [comment]
Excuse me,  I also want to ask, after upgrading the version， when status from TRIGGERED to OK,  I don't want it to notify， How can I ssachieve this?

In other words， I have judged whether there are new records in a certain table within 10 minutes. If so, I will send an alert email notification.

when status from TRIGGERED to OK， Prove that there are no new records，Then  shouldn’t send email。

Looking forward to hearing from you








________________________________
发件人: Zach Liu ***@***.***>
发送时间: 2024年9月10日 21:36
收件人: getredash/redash ***@***.***>
抄送: Yang Zhang @ Marsladder ***@***.***>; Comment ***@***.***>
主题: Re: [getredash/redash] when new Alert Destination email , Unable to set email content, Only set Subject Template ?????? (Discussion #7146)


ah right, i'm using the latest 24.09.0-dev

―

**Reply by zachliu** [reply]
i'm not sure if you can achieve this goal here. afaik, every status change must be notified:
* OK to TRIGGERED
* TRIGGERED to OK

**Reply by gaecoli** [reply]
> Excuse me, I also want to ask, after upgrading the version， when status from TRIGGERED to OK, I don't want it to notify， How can I ssachieve this? In other words， I have judged whether there are new records in a certain table within 10 minutes. If so, I will send an alert email notification. when status from TRIGGERED to OK， Prove that there are no new records，Then shouldn’t send email。 Looking forward to hearing from you
> […](#)
> ________________________________ 发件人: Zach Liu ***@***.***> 发送时间: 2024年9月10日 21:36 收件人: getredash/redash ***@***.***> 抄送: Yang Zhang @ Marsladder ***@***.***>; Comment ***@***.***> 主题: Re: [getredash/redash] when new Alert Destination email , Unable to set email content, Only set Subject Template ?????? (Discussion #7146) ah right, i'm using the latest 24.09.0-dev ― Reply to this email directly, view it on GitHub<[#7146 (reply in thread)](https://github.com/getredash/redash/discussions/7146#discussioncomment-10602804)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BG6RBRCD626FJHSXIIBDW7TZV3YVXAVCNFSM6AAAAABN4D3CCOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRQGI4DANA>. You are receiving this because you commented.Message ID: ***@***.***>

```
def should_notify(alert, new_state):
    passed_rearm_threshold = False
    if alert.rearm and alert.last_triggered_at:
        passed_rearm_threshold = (
            alert.last_triggered_at + datetime.timedelta(seconds=alert.rearm)
            < utils.utcnow()
        )

    return new_state != alert.state or (
        alert.state == models.Alert.TRIGGERED_STATE and passed_rearm_threshold
    )
```

This code is notify logistic. You can change code to fit your heart.

**Comment by ZhangYang19980825** [comment]
Sorry to bother you again，I upgraded redash to v10,   when custom template ，

In the following situations，

I can't get specific name value or email value in template


[cid:6e3fc9e2-21e2-479e-b7c1-861e01937cfa]

________________________________
发件人: Zach Liu ***@***.***>
发送时间: 2024年9月11日 22:17
收件人: getredash/redash ***@***.***>
抄送: Yang Zhang @ Marsladder ***@***.***>; Comment ***@***.***>
主题: Re: [getredash/redash] when new Alert Destination email , Unable to set email content, Only set Subject Template ?????? (Discussion #7146)


i'm not sure if you can achieve this goal here. afaik, every status change must be notified:

  *   OK to TRIGGERED
  *   TRIGGERED to OK

―

**Reply by zachliu** [reply]
i haven't used v10. can you take some screenshots?

**Reply by zachliu** [reply]
on `24.09.0-dev` i have

![2024-09-18_15-19](https://github.com/user-attachments/assets/b6ffe658-2bf2-429c-8797-f5581de2ff57)

![2024-09-18_15-19_1](https://github.com/user-attachments/assets/c7e1660a-7ae0-4641-9a4f-c883744806c0)
